libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -396,7 +399,7 @@ toCtx ::
   Var v ->
   Term fn a ->
   m (Ctx fn v a)
-toCtx v = go . simplifyTerm
+toCtx v = go -- . simplifyTerm
```

Comment:
It probably shouldn't be commented out, I can just remove it.

But basically the idea is that we try to reduce the number of calls to simplifyTerm as much as possible (because that in turn reduces the number of traversal of terms etc.) so we are now a but more careful about ensuring that terms are simplified before we end up computing contexts, for example.